File: docs/examples/1_login.cs
using System;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;

public class LoginExample {
    public static async Task Login() {
        using (var client = new HttpClient()) {
            client.BaseAddress = new Uri("http://localhost:3000");
            
            var content = new StringContent(
                JsonSerializer.Serialize({"email":"learner@example.com","password":"Password123!"}),
                Encoding.UTF8,
                "application/json"
            );

            var response = await client.postAsync("/auth/login", content);
            
            var result = await response.Content.ReadAsStringAsync();
            Console.WriteLine("Response: " + result);
        }
    }

    public static void Main() {
        Login().Wait();
    }
}

File: docs/examples/1_login.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
)

func Login() {
	client := &http.Client{}
	payload := map[string]interface{}{
		"email": "learner@example.com",
		"password": "Password123!"
	}
	body, _ := json.Marshal(payload)
	req, _ := http.NewRequest("POST", "http://localhost:3000/auth/login", bytes.NewBuffer(body))
	req.Header.Add("Content-Type", "application/json")

	resp, err := client.Do(req)
	if err != nil {
		fmt.Println("Error:", err)
		return
	}
	defer resp.Body.Close()

	body, _ := ioutil.ReadAll(resp.Body)
	var result interface{}
	json.Unmarshal(body, &result)
	fmt.Println("Response:", result)
}

func main() {
	login()
}